node/src/frontier_service.rs

```diff
@@ -90,8 +104,70 @@ pub fn spawn_frontier_tasks<B, BE, C>(
 	task_manager.spawn_essential_handle().spawn(
 		"frontier-fee-history",
 		Some("frontier"),
-		EthTask::fee_history_task(client, overrides, fee_history_cache, fee_history_cache_limit),
+		EthTask::fee_history_task(
+			client.clone(),
+			overrides.clone(),
+			fee_history_cache,
+			fee_history_cache_limit,
+		),
 	);
+
+	if eth_rpc_config.tracing_api.contains(&TracingApi::Debug)
+		|| eth_rpc_config.tracing_api.contains(&TracingApi::Trace)
+	{
+		let permit_pool = Arc::new(Semaphore::new(eth_rpc_config.tracing_max_permits as usize));
+		let (trace_filter_task, trace_filter_requester) =
+			if eth_rpc_config.tracing_api.contains(&TracingApi::Trace) {
+				let (trace_filter_task, trace_filter_requester) = CacheTask::create(
+					Arc::clone(&client),
+					Arc::clone(&backend),
+					Duration::from_secs(eth_rpc_config.tracing_cache_duration),
+					Arc::clone(&permit_pool),
+					Arc::clone(&overrides),
+				);
+				(Some(trace_filter_task), Some(trace_filter_requester))
+			} else {
+				(None, None)
+			};
+
+		let (debug_task, debug_requester) =
+			if eth_rpc_config.tracing_api.contains(&TracingApi::Debug) {
+				let (debug_task, debug_requester) = DebugHandler::task(
+					Arc::clone(&client),
+					Arc::clone(&backend),
+					Arc::clone(&frontier_backend),
+					Arc::clone(&permit_pool),
+					Arc::clone(&overrides),
+					eth_rpc_config.tracing_raw_max_memory_usage,
+				);
+				(Some(debug_task), Some(debug_requester))
+			} else {
+				(None, None)
+			};
+
+		// `trace_filter` cache task. Essential.
+		// Proxies rpc requests to it's handler.
+		if let Some(trace_filter_task) = trace_filter_task {
+			task_manager.spawn_essential_handle().spawn(
+				"trace-filter-cache",
+				Some("eth-tracing"),
+				trace_filter_task,
+			);
+		}
+
+		// `debug` task if enabled. Essential.
+		// Proxies rpc requests to it's handler.
+		if let Some(debug_task) = debug_task {
+			task_manager.spawn_essential_handle().spawn(
+				"tracing_api-debug",
+				Some("eth-tracing"),
+				debug_task,
+			);
+		}
+		RpcRequesters { debug: debug_requester, trace: trace_filter_requester }
```

Comment:
We can enable the debug and trace at the same time?
What differences between them?

Reply:
> We can enable the debug and trace at the same time?

Yes, we can. the debug and trace are two types of rpc, enable `debug_transaction`, `debug_blockByNumber` and `trace_filter` respectively. Check out the ts test in this pr.